core/src/main/java/hudson/PluginWrapper.java

```diff
@@ -214,27 +208,22 @@ public Dependency(String s) {
             if(idx==-1)
                 throw new IllegalArgumentException("Illegal dependency specifier "+s);
             this.shortName = s.substring(0,idx);
-            String version = s.substring(idx+1);
-
+            this.version = s.substring(idx+1);
+            
             boolean isOptional = false;
-            String[] osgiProperties = version.split("[;]");
+            String[] osgiProperties = s.split(";");
             for (int i = 1; i < osgiProperties.length; i++) {
                 String osgiProperty = osgiProperties[i].trim();
                 if (osgiProperty.equalsIgnoreCase("resolution:=optional")) {
                     isOptional = true;
                 }
             }
             this.optional = isOptional;
-            if (isOptional) {
-                this.version = osgiProperties[0];
-            } else {
-                this.version = version;
-            }
```

Comment:
Note that this part was actually a bug fix, and should be safe to keep. But yeah it can be reinstated later.